### ISSUE TITLE
Too low loss and Too poor prediction 

### ISSUE THREAD

**Issue by oglee815** [ibm-granite/granite-tsfm#73]
Hi, I am trying to train TTM from scratch on some stock price dataset(more than 200K samples, without almost no code modification except dataset load part)
But it is strange.

Training loss is started with relatively very low value, like 0.000xx,  and quickly converged.
And when I plot the prediction values with test set, it looks very very flat linear line.
It seems not trained at all.

Any suggestion to make modification to fit on some new dataset? 😢

**Comment by oglee815** [comment]
My data looks like this:

	candle_date_time_kst	opening_price	high_price	low_price	 trade_price
0	2017-09-25 12:00:00	4201000.0	4220000.0	4195000.0	4216000.0
1	2017-09-25 12:16:00	4222000.0	4244000.0	4210000.0	4229000.0
2	2017-09-25 12:32:00	4219000.0	4225000.0	4204000.0	4204000.0
3	2017-09-25 12:50:00	4215000.0	4215000.0	4206000.0	4206000.0
4	2017-09-25 12:01:00	4213000.0	4222000.0	4204000.0	4204000.0

the values approximately have mean of 2.911540e+07, std of 2.309221e+07
total size = 348133,
first epoch train loss = 0.000900
first epoch validation loss = 0.000482

But, when I use single target columns, train loss becomes more normal (0.xx)

**Comment by david-yang-tfs** [comment]
the flat linear line is a symptom of the data not being time series model able, in my experience. it tracks with you working with a stock market dataset

**Comment by flbyrne** [comment]
@oglee815 I'm also training several target columns (~400) but the losses look ok and takes several epochs to converge. However, when running inference, I wasn't able to get a prediction for all the target columns at the same time and had to train the pre-processor for each column and forecast for each column individually. Did you have that problem?

**Comment by francoisailah** [comment]
@oglee815 
Could you please how you are training ttm from scratch (codes)?

**Comment by flbyrne** [comment]
I'm not training ttm from scratch, just finetuning.

**Comment by aixuedegege** [comment]
> Hi, I am trying to train TTM from scratch on some stock price dataset(more than 200K samples, without almost no code modification except dataset load part) But it is strange.
> 
> Training loss is started with relatively very low value, like 0.000xx, and quickly converged. And when I plot the prediction values with test set, it looks very very flat linear line. It seems not trained at all.
> 
> Any suggestion to make modification to fit on some new dataset? 😢

did you solve it? how long do you predict when trainning ?
